relay-general/src/processor/selector.rs

```diff
@@ -138,22 +138,24 @@ pub enum SelectorSpec {
     Or(Vec<SelectorSpec>),
     Not(Box<SelectorSpec>),
     Path(Vec<SelectorPathItem>),
+    Other(String),
 }
 
 impl fmt::Display for SelectorSpec {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
+        match self {
             SelectorSpec::And(ref xs) => {
                 for (idx, x) in xs.iter().enumerate() {
                     if idx > 0 {
                         write!(f, " && ")?;
                     }
 
-                    let needs_parens = match *x {
+                    let needs_parens = match x {
                         SelectorSpec::And(_) => false,
                         SelectorSpec::Or(_) => true,
                         SelectorSpec::Not(_) => false,
                         SelectorSpec::Path(_) => false,
+                        SelectorSpec::Other(_) => true,
```

Comment:
figured it's better with too many parens than too few

Reply:
I think having parenthesis is required if we think about forward compatibility.